scripts/download_technique_animations.py

```diff
@@ -8,67 +8,126 @@
+                # Assemble credits and skip the "Category:" blocks   
+                comment_tail: str = comment_entry.tail
+                if comment_tail is not None and not comment_tail.startswith("Category:"):
+                    credits_text += comment_entry.tail.strip()
+
+    credits_text = credits_text or "Unknown animation source/author"
+    credits_record: str = CREDITS_TEMPLATE.format(animation_name=animation_name, animation_url=gif_page_url, credits_text=credits_text)
+    return credits_record
+
+
+def gif_to_frames(filepath: str) -> None:
     """Extract individual animation frames as PNG from a GIF."""
-    with Image.open(filename) as image:
+    with Image.open(filepath) as image:
         if not image.is_animated:
-            print(f"{filename} is not animated, skipped")
+            print(f"{filepath} is not animated, skipped")
             return
         if not image.width == image.height == FRAME_SIZE:
-            print(f"{filename} is not {FRAME_SIZE}x{FRAME_SIZE}, skipped")
+            print(f"{filepath} is not {FRAME_SIZE}x{FRAME_SIZE}, skipped")
             return
-        base_name = process_filename(filename)
+  
+        base_name: str = process_filename(filepath)
+        animation_name: str = process_animation_name(base_name)
         for frame in range(0, image.n_frames):
-            frame_filename = f"{base_name}{frame:02}.png"
-            print(f"saving {filename}:{frame_filename} {frame}/{image.n_frames - 1}")
+            frame_filename: str = os.path.join(ANIMATION_DIR, f"{base_name}{frame:02}.png")
+            print(f"Generating animation frames for '{animation_name}': {frame_filename} - {frame}/{image.n_frames - 1}")
             image.seek(frame)
             image.save(frame_filename, optimize=True)
 
 
 # TODO: Is passing the Wiki URL needed?
 def download_technique_animations(wiki_url: str) -> None:
     """Download technique animation frames from the Tuxemon Wiki."""
-    print(f"Getting page source for URL: {wiki_url}")
-    source = requests.get(wiki_url)
-    tree = html.fromstring(source.content)
-    elements = tree.xpath("//li[@class='gallerybox']//img")
-
-    for index, element in enumerate(elements, start=1):
-        url = urljoin(wiki_url, element.get("src"))
-        filename = url.split('/')[-1]
-        print(f"{index}/{len(elements)} downloading {filename}")
-        download_bytes(url, filename)
-        process_gif(filename)
+    print(f"Getting animations and metadata from URL: {wiki_url}")
+
+    # Animation GIF path
+    animations_url = f"{wiki_url}/index.php?title=Category:Technique_Animation"
+    anim_source = requests.get(animations_url)
+    anim_tree = html.fromstring(anim_source.content)
+
+    with tempfile.TemporaryDirectory() as tmpdirname:
+
+        elements = anim_tree.xpath("//li[@class='gallerybox']//a[@class='image']")
+        with open(CREDITS_FILENAME, "w") as credits_file:
+            print("### Technique Animations", file=credits_file)
```

Comment:
Using `file.write()` or `file.writelines()` excludes the terminal endline symbol.